test/local/account_routes.js

```diff
@@ -251,101 +250,139 @@ test('/account/reset', function (t) {
+    device.id = deviceId.toString('hex')
+
+    t.test('identical data', function (t) {
+      mockRequest.auth.credentials.deviceId = deviceId
+      mockRequest.payload.name = 'my awesome device'
 
-    return runTest(route, mockRequest, function (response) {
-      t.equal(mockLog.increment.callCount, 0, 'log.increment was not called')
+      return runTest(route, mockRequest, function (response) {
+        t.equal(mockDB.updateDevice.callCount, 0, 'updateDevice was not called')
 
-      t.equal(mockDevices.upsert.callCount, 1, 'devices.upsert was called once')
-      var args = mockDevices.upsert.args[0]
-      t.equal(args[2].id, mockRequest.auth.credentials.deviceId.toString('hex'), 'payload.id defaulted to credentials.deviceId')
-    })
-    .then(function () {
-      mockLog.increment.reset()
-      mockDevices.upsert.reset()
+        t.equal(mockLog.increment.callCount, 1, 'a counter was incremented')
+        t.equal(mockLog.increment.firstCall.args[0], 'device.update.spurious')
+
+        t.deepEqual(response, mockRequest.payload)
+      })
+      .then(function () {
+        mockLog.increment.reset()
+      })
     })
-  }, t)
 
-  t.test('device updates disabled', function (t) {
-    config.deviceUpdatesEnabled = false
+    t.test('different data', function (t) {
+      mockRequest.auth.credentials.deviceId = crypto.randomBytes(16)
+      var payload = mockRequest.payload
+      payload.name = 'my even awesomer device'
+      payload.type = 'phone'
+      payload.pushCallback = 'https://push.services.mozilla.com/123456'
+      payload.pushPublicKey = 'SomeEncodedBinaryStuffThatDoesntGetValidedByThisTest'
 
-    return runTest(route, mockRequest, function () {
-      t.fail('should have thrown')
+      return runTest(route, mockRequest, function (response) {
+        t.equal(mockDB.updateDevice.callCount, 1, 'updateDevice was called')
+
+        t.equal(mockLog.activityEvent.callCount, 1, 'log.activityEvent was called once')
+        var args = mockLog.activityEvent.args[0]
+        t.equal(args.length, 3, 'log.activityEvent was passed three arguments')
+        t.equal(args[0], 'device.updated', 'first argument was event name')
+        t.equal(args[1], mockRequest, 'second argument was request object')
+        t.deepEqual(args[2], { uid: uid.toString('hex'), device_id: deviceId.toString('hex') }, 'third argument contained uid')
+
+        t.equal(mockLog.notifyAttachedServices.callCount, 0, 'log.notifyAttachedServices was not called')
+
+        t.equal(mockLog.increment.callCount, 5, 'the counters were incremented')
+        t.equal(mockLog.increment.getCall(0).args[0], 'device.update.sessionToken')
+        t.equal(mockLog.increment.getCall(1).args[0], 'device.update.name')
+        t.equal(mockLog.increment.getCall(2).args[0], 'device.update.type')
+        t.equal(mockLog.increment.getCall(3).args[0], 'device.update.pushCallback')
+        t.equal(mockLog.increment.getCall(4).args[0], 'device.update.pushPublicKey')
+      })
```

Comment:
Make sense to reset `mockDB.updateDevice`, `mockLog.activityEvent.callCount` and `mockLog.increment` since they have been called?